In [1]:
import pandas as pd

Los objetivos de la clase de hoy son:

    1. Cargad el csv que guardasteis ayer.
Antes de seguir con las preguntas, para evitar posibles errores con los nulos vamos a eliminar todos los nulos del DataFrame, esto lo haremos sobre el archivo de 2016.xlsx. Para ello vamos a utilizar el método .dropna() de Pandas. Tendréis que ejecutar la siguiente línea de código para eliminar los nulos del DataFrame y poder seguir haciendo el resto de los ejercicios.
df.dropna(inplace = True)


In [10]:
df = pd.read_csv("datos/2016_edit.csv")


In [11]:
df.shape

(13051, 10)

In [12]:
df.dropna(inplace = True)

In [13]:
df.shape

(12939, 10)


        1.1 ¿Cuántas filas se han eliminado con esta línea de código?

Se han eliminado 112 filas

In [25]:
df.tail(20)

id_ticket       fecha  id_empleado  id_agente      categoría  \
13031  TWRESR-7342556289  2016-07-05        188.0       32.0        Sistema   
13032  TWRESR-7742461447  2016-04-01       1734.0       14.0        Sistema   
13033  TWRESR-9242723942  2016-12-19       1964.0       19.0        Sistema   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0       Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0       Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0       Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0       Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0       Hardware   
13039  TWRTNR-0042611797  2016-08-29       1549.0       37.0  Acceso&%Login   
13040  TWRTNR-3042614921  2016-09-01        582.0       19.0  Acceso&%Login   
13041  TWRTNR-8142621481  2016-09-08       1848.0       24.0  Acceso&%Login   
13042  TWRTSR-0042500565  2016-05-10        866.0       15.0        Sistema   
13043  TWRTSR-0242467139  2016-04-07        643.0       31.0        Sistema   
13044  TWRTSR-2642729724  2016-12-25       1352.0        7.0        Sistema   
13045  TWRTSR-4642439902  2016-03-10        750.0       19.0        Sistema   
13046  TWRTSR-6442574526  2016-07-23        962.0       25.0        Sistema   
13047  TWRTSR-7142570866  2016-07-19        126.0       38.0        Sistema   
13048  TWRTSR-7742498308  2016-05-08        510.0        3.0        Sistema   
13049  TWRTSR-8642558628  2016-07-07       1942.0       26.0        Sistema   
13050  TWRTSR-8742628897  2016-09-15        919.0        8.0        Sistema   

            tipo  severidad prioridad  días_resolución  satisfacción  
13031   Problema  3 - Mayor  1 - Baja              3.0           4.0  
13032   Problema  3 - Mayor  1 - Baja              7.0           5.0  
13033   Problema  1 - Menor  1 - Baja              2.0           4.0  
13034  Solicitud  1 - Menor  1 - Baja             10.0           4.0  
13035  Solicitud  1 - Menor  1 - Baja             10.0           4.0  
13036  Solicitud  1 - Menor  1 - Baja              7.0           5.0  
13037  Solicitud  1 - Menor  1 - Baja              5.0           5.0  
13038  Solicitud  3 - Mayor  1 - Baja              7.0           5.0  
13039  Solicitud  3 - Mayor  1 - Baja              0.0           4.0  
13040  Solicitud  1 - Menor  1 - Baja              0.0           4.0  
13041  Solicitud  3 - Mayor  1 - Baja              1.0           5.0  
13042  Solicitud  1 - Menor  1 - Baja             10.0           5.0  
13043  Solicitud  3 - Mayor  1 - Baja              6.0           1.0  
13044  Solicitud  1 - Menor  1 - Baja             12.0           5.0  
13045  Solicitud  1 - Menor  1 - Baja             16.0           5.0  
13046  Solicitud  1 - Menor  1 - Baja             14.0           4.0  
13047  Solicitud  3 - Mayor  1 - Baja             13.0           5.0  
13048  Solicitud  3 - Mayor  1 - Baja              2.0           5.0  
13049  Solicitud  1 - Menor  1 - Baja             16.0           4.0  
13050  Solicitud  1 - Menor  1 - Baja              7.0           4.0

    2. ¿Cuál es el número de incidencias que se han dado de tipo "Sistema", lo encontraréis en la columna de categoría?


In [18]:
n_incidencias = df[df["categoría"] == "Sistema"]
len(n_incidencias)

5211

Hay 5211 incidencias de Sistema.

    3. ¿Y para la categorías de "Hardware" y "Software"? Cread un dataframe nuevo con los resultados de este filtrado de datos. 📌 Nota Este ejercicio se puede resolver usando dos condiciones de filtrado o usando el método isin(). Hacedlo con ambas herramientas.


In [34]:
df_hard_soft = df[(df["categoría"] == "Hardware") |  (df["categoría"] == "Software")]
df_hard_soft

id_ticket       fecha  id_empleado  id_agente categoría  \
4      GDLEER-0042524120  2016-06-03       1842.0       31.0  Software   
5      GDLEER-0142608095  2016-08-26         59.0       20.0  Software   
6      GDLEER-0242564650  2016-07-13       1175.0       36.0  Software   
7      GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8      GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
...                  ...         ...          ...        ...       ...   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0  Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0  Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0  Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0  Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  
4       Problema  2 - Normal  0 - Sin Asignar              0.0           5.0  
5       Problema  2 - Normal  0 - Sin Asignar              1.0           1.0  
6       Problema  2 - Normal  0 - Sin Asignar              2.0           1.0  
7       Problema  2 - Normal  0 - Sin Asignar              5.0           5.0  
8       Problema  2 - Normal  0 - Sin Asignar              8.0           5.0  
...          ...         ...              ...              ...           ...  
13034  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13035  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13036  Solicitud   1 - Menor         1 - Baja              7.0           5.0  
13037  Solicitud   1 - Menor         1 - Baja              5.0           5.0  
13038  Solicitud   3 - Mayor         1 - Baja              7.0           5.0  

[3846 rows x 10 columns]

In [48]:
df_hs = df[df["categoría"].isin(["Hardware", "Software"])]
df_hs

id_ticket       fecha  id_empleado  id_agente categoría  \
4      GDLEER-0042524120  2016-06-03       1842.0       31.0  Software   
5      GDLEER-0142608095  2016-08-26         59.0       20.0  Software   
6      GDLEER-0242564650  2016-07-13       1175.0       36.0  Software   
7      GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8      GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
...                  ...         ...          ...        ...       ...   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0  Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0  Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0  Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0  Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  
4       Problema  2 - Normal  0 - Sin Asignar              0.0           5.0  
5       Problema  2 - Normal  0 - Sin Asignar              1.0           1.0  
6       Problema  2 - Normal  0 - Sin Asignar              2.0           1.0  
7       Problema  2 - Normal  0 - Sin Asignar              5.0           5.0  
8       Problema  2 - Normal  0 - Sin Asignar              8.0           5.0  
...          ...         ...              ...              ...           ...  
13034  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13035  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13036  Solicitud   1 - Menor         1 - Baja              7.0           5.0  
13037  Solicitud   1 - Menor         1 - Baja              5.0           5.0  
13038  Solicitud   3 - Mayor         1 - Baja              7.0           5.0  

[3846 rows x 10 columns]

In [35]:
df_hard_soft

id_ticket       fecha  id_empleado  id_agente categoría  \
4      GDLEER-0042524120  2016-06-03       1842.0       31.0  Software   
5      GDLEER-0142608095  2016-08-26         59.0       20.0  Software   
6      GDLEER-0242564650  2016-07-13       1175.0       36.0  Software   
7      GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8      GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
...                  ...         ...          ...        ...       ...   
13034  TWRTER-2042528927  2016-06-07        462.0       19.0  Software   
13035  TWRTER-5042710887  2016-12-06       1098.0       38.0  Hardware   
13036  TWRTER-8742418826  2016-02-18        942.0       28.0  Software   
13037  TWRTER-9542402645  2016-02-02        334.0       26.0  Software   
13038  TWRTER-9642514691  2016-05-24        909.0       16.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  
4       Problema  2 - Normal  0 - Sin Asignar              0.0           5.0  
5       Problema  2 - Normal  0 - Sin Asignar              1.0           1.0  
6       Problema  2 - Normal  0 - Sin Asignar              2.0           1.0  
7       Problema  2 - Normal  0 - Sin Asignar              5.0           5.0  
8       Problema  2 - Normal  0 - Sin Asignar              8.0           5.0  
...          ...         ...              ...              ...           ...  
13034  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13035  Solicitud   1 - Menor         1 - Baja             10.0           4.0  
13036  Solicitud   1 - Menor         1 - Baja              7.0           5.0  
13037  Solicitud   1 - Menor         1 - Baja              5.0           5.0  
13038  Solicitud   3 - Mayor         1 - Baja              7.0           5.0  

[3846 rows x 10 columns]

In [45]:
df_hard_soft.tipo.unique()

array(['Problema', 'Solicitud'], dtype=object)

        3.1 De los datos filtrados en el punto 3. ¿Cuántas de ellas son de tipo "Solicitud" y cuántas de tipo "Problema"?


In [79]:
df_solicitud = df_hard_soft[df_hard_soft["tipo"] == "Solicitud"]
df_solicitud.shape[0]


2876

In [81]:
df_problema = df_hard_soft[df_hard_soft["tipo"] == "Problema"]
df_problema.shape[0]

970

        3.2 ¿Cuántas solicitudes tienen una satisfacción de más de 3?


In [82]:
df_sol_plus = df_solicitud[df_solicitud["satisfacción"] > 3]
df_sol_plus.shape[0]

2251

In [49]:
df_hard_soft["satisfacción"].unique()

array([5., 1., 4., 3., 2.])

In [83]:
df_sol_plus["satisfacción"].unique() 

array([5., 4.])

In [54]:
df_sol_plus.head()

id_ticket       fecha  id_empleado  id_agente categoría  \
249  GDLTER-0142410420  2016-02-10       1172.0       34.0  Hardware   
250  GDLTER-0142541943  2016-06-20        434.0       29.0  Hardware   
251  GDLTER-0142627596  2016-09-14       1489.0       15.0  Hardware   
252  GDLTER-0242370808  2016-01-01        920.0        8.0  Software   
253  GDLTER-0342596564  2016-08-14       1196.0       15.0  Software   

          tipo   severidad        prioridad  días_resolución  satisfacción  
249  Solicitud  2 - Normal  0 - Sin Asignar             11.0           5.0  
250  Solicitud  2 - Normal  0 - Sin Asignar             11.0           5.0  
251  Solicitud  2 - Normal  0 - Sin Asignar             11.0           5.0  
252  Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
253  Solicitud  2 - Normal  0 - Sin Asignar              3.0           5.0

        3.3 ¿Y cuántas tienen una satisfacción mayor o igual que 4 y han tardado menos de 3 días de resolución?


In [59]:
cond1 = df_sol_plus["satisfacción"] >=4
cond2 = df_sol_plus["días_resolución"] < 3

df_sol_premiun = df_sol_plus[cond1 & cond2]
df_sol_premiun

id_ticket       fecha  id_empleado  id_agente categoría  \
252    GDLTER-0242370808  2016-01-01        920.0        8.0  Software   
268    GDLTER-1242412486  2016-02-12        578.0       34.0  Software   
270    GDLTER-1242699033  2016-11-25       1453.0       30.0  Software   
306    GDLTER-3242386725  2016-01-17       1072.0       17.0  Software   
309    GDLTER-3242498568  2016-05-08       1206.0       35.0  Software   
...                  ...         ...          ...        ...       ...   
12768  TWLTER-6242650232  2016-10-07       1313.0        2.0  Software   
12774  TWLTER-7242522978  2016-06-01        717.0       49.0  Hardware   
12790  TWLTER-8542726462  2016-12-22       1986.0       24.0  Hardware   
12792  TWLTER-8942708246  2016-12-04       1364.0       22.0  Hardware   
12794  TWLTER-9342575845  2016-07-24        774.0       38.0  Hardware   

            tipo   severidad        prioridad  días_resolución  satisfacción  
252    Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
268    Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
270    Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
306    Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
309    Solicitud  2 - Normal  0 - Sin Asignar              2.0           5.0  
...          ...         ...              ...              ...           ...  
12768  Solicitud  2 - Normal         1 - Baja              2.0           5.0  
12774  Solicitud  2 - Normal         1 - Baja              1.0           4.0  
12790  Solicitud  2 - Normal         1 - Baja              1.0           4.0  
12792  Solicitud  2 - Normal         1 - Baja              2.0           5.0  
12794  Solicitud  2 - Normal         1 - Baja              0.0           4.0  

[285 rows x 10 columns]

    4. Ahora nuestro jefe quiere saber que cantidad de tickets ha habido en la segunda quincena de Enero. Cread un DataFrame nuevo con los tickets correspondientes a la segunda quincena de Enero.

In [64]:
df.head()

id_ticket       fecha  id_empleado  id_agente      categoría  \
0  GDDENR-5042564453  2016-07-13       1735.0        4.0  Acceso&%Login   
1  GDDENR-8042508060  2016-05-18       1566.0       10.0  Acceso&%Login   
2  GDDESR-1342539995  2016-06-18        569.0       29.0        Sistema   
3  GDDTSR-5942488006  2016-04-28        320.0       40.0        Sistema   
4  GDLEER-0042524120  2016-06-03       1842.0       31.0       Software   

        tipo           severidad        prioridad  días_resolución  \
0   Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
1   Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
2   Problema  0 - Sin Clasificar  0 - Sin Asignar              3.0   
3  Solicitud  0 - Sin Clasificar  0 - Sin Asignar              9.0   
4   Problema          2 - Normal  0 - Sin Asignar              0.0   

   satisfacción  
0           5.0  
1           5.0  
2           5.0  
3           5.0  
4           5.0

In [72]:
df.info(
    
)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12939 entries, 0 to 13050
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id_ticket        12939 non-null  object 
 1   fecha            12939 non-null  object 
 2   id_empleado      12939 non-null  float64
 3   id_agente        12939 non-null  float64
 4   categoría        12939 non-null  object 
 5   tipo             12939 non-null  object 
 6   severidad        12939 non-null  object 
 7   prioridad        12939 non-null  object 
 8   días_resolución  12939 non-null  float64
 9   satisfacción     12939 non-null  float64
dtypes: float64(4), object(6)
memory usage: 1.1+ MB


In [75]:
df_2enero= df[(df["fecha"] >= "2016-01-15") & (df["fecha"] <= "2016-01-31")]
df_2enero.head()

id_ticket       fecha  id_empleado  id_agente      categoría  \
12  GDLEER-2142385152  2016-01-16       1825.0       31.0       Software   
65  GDLEET-8142386963  2016-01-17        376.0       39.0       Software   
68  GDLEET-8942397489  2016-01-28        488.0       14.0       Hardware   
70  GDLEET-9342397848  2016-01-28        104.0       48.0       Software   
74  GDLENR-1042392759  2016-01-23       1755.0       17.0  Acceso&%Login   

        tipo   severidad        prioridad  días_resolución  satisfacción  
12  Problema  2 - Normal  0 - Sin Asignar              1.0           5.0  
65  Problema  2 - Normal  0 - Sin Asignar              1.0           5.0  
68  Problema  2 - Normal  0 - Sin Asignar              9.0           5.0  
70  Problema  2 - Normal  0 - Sin Asignar              3.0           5.0  
74  Problema  2 - Normal  0 - Sin Asignar              0.0           5.0

    4.1 ¿Ha habido algun ticket en esta quincena que corresponda a la categoría de Solicitud?

In [95]:
df_2enero["tipo"].unique()

array(['Problema', 'Solicitud'], dtype=object)

SI

        4.2 Revisando los datos, nuestro jefe se ha dado cuenta que hay muchos tickets no tienen ninguna prioridad asignada. Esto no debería seguir repitiéndose ya que cada ticket debe tener una prioridad para saber que urgencia debemos darle a cada ticket. Usando el str.contains extraed los tickets que no tengan una prioridad asignada o que tengan una prioridad baja.


In [85]:
import re

In [84]:
df.prioridad.unique()

array(['0 - Sin Asignar', '3 - Alta', '2 - Media', '1 - Baja'],
      dtype=object)

In [90]:
patron =  "^[0|1].*"

df_prioridad = df[df.prioridad.str.contains(patron, regex=True)]

In [92]:
df_prioridad

id_ticket       fecha  id_empleado  id_agente      categoría  \
0      GDDENR-5042564453  2016-07-13       1735.0        4.0  Acceso&%Login   
1      GDDENR-8042508060  2016-05-18       1566.0       10.0  Acceso&%Login   
2      GDDESR-1342539995  2016-06-18        569.0       29.0        Sistema   
3      GDDTSR-5942488006  2016-04-28        320.0       40.0        Sistema   
4      GDLEER-0042524120  2016-06-03       1842.0       31.0       Software   
...                  ...         ...          ...        ...            ...   
13046  TWRTSR-6442574526  2016-07-23        962.0       25.0        Sistema   
13047  TWRTSR-7142570866  2016-07-19        126.0       38.0        Sistema   
13048  TWRTSR-7742498308  2016-05-08        510.0        3.0        Sistema   
13049  TWRTSR-8642558628  2016-07-07       1942.0       26.0        Sistema   
13050  TWRTSR-8742628897  2016-09-15        919.0        8.0        Sistema   

            tipo           severidad        prioridad  días_resolución  \
0       Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
1       Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
2       Problema  0 - Sin Clasificar  0 - Sin Asignar              3.0   
3      Solicitud  0 - Sin Clasificar  0 - Sin Asignar              9.0   
4       Problema          2 - Normal  0 - Sin Asignar              0.0   
...          ...                 ...              ...              ...   
13046  Solicitud           1 - Menor         1 - Baja             14.0   
13047  Solicitud           3 - Mayor         1 - Baja             13.0   
13048  Solicitud           3 - Mayor         1 - Baja              2.0   
13049  Solicitud           1 - Menor         1 - Baja             16.0   
13050  Solicitud           1 - Menor         1 - Baja              7.0   

       satisfacción  
0               5.0  
1               5.0  
2               5.0  
3               5.0  
4               5.0  
...             ...  
13046           4.0  
13047           5.0  
13048           5.0  
13049           4.0  
13050           4.0  

[6016 rows x 10 columns]

In [91]:
df_prioridad.prioridad.unique()

array(['0 - Sin Asignar', '1 - Baja'], dtype=object)

    5. Nuestro jefe quiere seguir buscando patrones y quiere saber si existe una relación entre el número de días que tarde en cerrarse un ticket y la satisfacción de los clientes. Para eso nos pide que:
        5.1 Extraigamos los tickets en lo que pasaron 5 o más días en ser solucionados.


In [96]:
df.head()

id_ticket       fecha  id_empleado  id_agente      categoría  \
0  GDDENR-5042564453  2016-07-13       1735.0        4.0  Acceso&%Login   
1  GDDENR-8042508060  2016-05-18       1566.0       10.0  Acceso&%Login   
2  GDDESR-1342539995  2016-06-18        569.0       29.0        Sistema   
3  GDDTSR-5942488006  2016-04-28        320.0       40.0        Sistema   
4  GDLEER-0042524120  2016-06-03       1842.0       31.0       Software   

        tipo           severidad        prioridad  días_resolución  \
0   Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
1   Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
2   Problema  0 - Sin Clasificar  0 - Sin Asignar              3.0   
3  Solicitud  0 - Sin Clasificar  0 - Sin Asignar              9.0   
4   Problema          2 - Normal  0 - Sin Asignar              0.0   

   satisfacción  
0           5.0  
1           5.0  
2           5.0  
3           5.0  
4           5.0

In [100]:
df_5dias = df[df["días_resolución"] >= 5]
df_5dias["días_resolución"].unique()

array([ 9.,  5.,  8., 10.,  6.,  7., 13., 16., 17., 11., 12., 14., 15.,
       19., 18., 21.])

In [102]:
df_5dias["satisfacción"].unique()

array([5., 1., 4., 3., 2.])

        5.2 Extraigamos los tickets en los que pasaron menos de 5 días en ser solucionados.


In [101]:
df_pocos_dias = df[df["días_resolución"] < 5]
df_pocos_dias["días_resolución"].unique()

array([0., 3., 1., 2., 4.])

In [107]:
df_pocos_dias["satisfacción"].unique()

array([5., 1., 4., 3., 2.])

        5.3 Usando como referencia los dos conjuntos de datos creados en los dos pasos anteriores:
            5.3.1. ¿Qué valores tenemos de satisfacción en cada uno de ellos?


In [108]:
df_pocos_dias["satisfacción"].unique()

array([5., 1., 4., 3., 2.])

            5.3.2. ¿En cuál de ellos hay más puntuaciones de 5 en relación a la satisfacción?


In [110]:
df_pocos_dias5 = df_pocos_dias[df_pocos_dias["satisfacción"] == 5]
df_pocos_dias5.shape[0]

3320

In [118]:
df_pocos_dias.head()

id_ticket       fecha  id_empleado  id_agente      categoría  \
0  GDDENR-5042564453  2016-07-13       1735.0        4.0  Acceso&%Login   
1  GDDENR-8042508060  2016-05-18       1566.0       10.0  Acceso&%Login   
2  GDDESR-1342539995  2016-06-18        569.0       29.0        Sistema   
4  GDLEER-0042524120  2016-06-03       1842.0       31.0       Software   
5  GDLEER-0142608095  2016-08-26         59.0       20.0       Software   

       tipo           severidad        prioridad  días_resolución  \
0  Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
1  Problema  0 - Sin Clasificar  0 - Sin Asignar              0.0   
2  Problema  0 - Sin Clasificar  0 - Sin Asignar              3.0   
4  Problema          2 - Normal  0 - Sin Asignar              0.0   
5  Problema          2 - Normal  0 - Sin Asignar              1.0   

   satisfacción  
0           5.0  
1           5.0  
2           5.0  
4           5.0  
5           1.0

In [111]:
df_5dias5 = df_5dias[df_5dias["satisfacción"] == 5]
df_5dias5.shape[0]

2883

In [117]:
df_5dias.head()

id_ticket       fecha  id_empleado  id_agente categoría  \
3   GDDTSR-5942488006  2016-04-28        320.0       40.0   Sistema   
7   GDLEER-0542574815  2016-07-23        561.0       18.0  Software   
8   GDLEER-0842457219  2016-03-28         71.0       12.0  Software   
15  GDLEER-2342731423  2016-12-27        422.0       24.0  Hardware   
18  GDLEER-3042628638  2016-09-15       1863.0       46.0  Hardware   

         tipo           severidad        prioridad  días_resolución  \
3   Solicitud  0 - Sin Clasificar  0 - Sin Asignar              9.0   
7    Problema          2 - Normal  0 - Sin Asignar              5.0   
8    Problema          2 - Normal  0 - Sin Asignar              8.0   
15   Problema          2 - Normal  0 - Sin Asignar             10.0   
18   Problema          2 - Normal  0 - Sin Asignar              9.0   

    satisfacción  
3            5.0  
7            5.0  
8            5.0  
15           5.0  
18           5.0

    6. Revisando los dataframes generados en el ejercicio anterior, nos hemos dado cuenta que hay una serie de empleados que tardan demasiado en resolver los problemas planteados por los clientes. Nuestro jefe quiere extraer los datos de estos empleados en concreto para saber que esta pasando. En concreto estos empleados son aquellos que tienen los siguientes id's: 1118.0, 730.0, 97.0, 1745.0. Crea un dataframe donde solo tengamos la información de estos empleados. Usad el método isin(), para resolver este ejercicio.


    7. Utilizando la columna de fecha cread una nueva columna llamada semestre que contenga 1er semestre o 2º semestre. Aquellos tickets que pertenezcan a los seis primeros meses tendran el valor de "1er trismestre" y los tickets de julio en adelante tendrán el valor de "2º semestre".


    8. Guardad como un archivo csv nuevo. Lo usaremos en el pair programming de mañana.